slackbridge/bots.py

```diff
@@ -229,17 +227,21 @@ def signedOn(self):
         # And if not, register for the first time
         self.msg('NickServ', 'GROUP {} {}'.format(self.target_group_nick,
                                                   self.nickserv_password))
-        for channel in self.channels:
-            self.log(log.msg, 'Joining #{}'.format(channel['name']))
-            self.join_channel(channel['name'])
+        for channel_name in self.joined_channels:
+            self.log(log.msg, 'Joining #{}'.format(channel_name))
+            self.join_channel(channel_name)
 
         self.away('Default away for startup.')
 
     def join_channel(self, channel_name):
         self.join('#{}'.format(channel_name))
+        if channel_name not in self.joined_channels:
+            self.joined_channels.append(channel_name)
 
     def part_channel(self, channel_name):
         self.leave('#{}'.format(channel_name))
```

Comment:
Should we check if the channel is in ```joined_channels``` before calling the ```leave``` method? Or could this lead to unexpected behavior if we, for example, joined and parted a channel in quick succession?

Reply:
Twisted also has a ```joined``` method that might be worth looking into. It's called when a user finishes joining a channel. That seems like a more logical place to append to ```joined_channels```.

Reply:
Given `leave` just calls `PART` under the hood, I don't think it really matters that much. I think if the bot isn't in the channel but tries to leave, it won't cause any problems, although I'd need to actually test that. The `joined_channel` stuff is really just an optimization right now anyway, so that all the bots don't join all the IRC channels. I do think your suggestion is fine, my main concern is that the `joined_channels` list gets out of date and then a bot can't leave (or worse, join) a channel any more.